doc/en/source/ngsildbasediotintegration.rst

```diff
@@ -0,0 +1,476 @@
+
+.. code-block:: console 
+
+	curl -L -X PATCH 'http://localhost:1026/ngsi-ld/v1/entities/urn:ngsi-ld:Device:water001/attrs/on' \
+	-H 'fiware-service: openiot' \
+	-H 'fiware-servicepath: /' \
+	-H 'Accept: application/ld+json' \
+	-H 'Link: <https://fiware.github.io/data-models/context.jsonld>; rel="https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld"; type="application/ld+json"' \
+	-H 'Content-Type: application/json' \
+	--data-raw '{
+
+        	"type": "Property",
+        	"value": " "
+
+	}'
+	
+**Step7** Check if FogFlow receives the subscribed entity. 
+-------------------------------------------------------------------
+
+Use the CURL command to query entities of type "Device" from  FogFlow thinBroker. 
+
+
+.. code-block:: console    
+
+	curl -iX GET \
+		  'http://localhost:8070/ngsi-ld/v1/entities?type=Device' \
+		  -H 'Content-Type: application/json' \
+		  -H 'Accept: application/ld+json' \
+		  -H 'fiware-service: openiot' \
+		  -H 'fiware-servicepath: /' \
+		  -H 'Link: <https://fiware.github.io/data-models/context.jsonld>; rel="https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld"; type="application/ld+json"' 
+
+Note: Replace localhost with the IP where fogflow thinbroker is running.
+Note: Replace the localhost with IP where Orion-LD broker is running and <fogflow_broker_IP> with the IP where the fogflow broker is running.
+
+
+How to Program and Apply a Data Analytics Function 
+================================================================
+
+**Step8** Please refer the steps below, to register fogfunction using dashboard.
+------------------------------------------------------------------------------
+
+1. To register Operator, open fogflow dashboard. Select Operator Registry Tab from horizontal bar, select operator from menu on left and then click register button. Right click on workspace and select operator from drop down list and enter details as shown and at last click on submit.
+
+
+.. figure:: figures/water_op.png
+
+
+2. Register the following docker image, corresponding to the operator created in above step.
+   
+   
+.. code-block:: console
+
+	fogflow/waterpump
+
+        The above operator send a "off" command after 1 hour if waterPump is "on".
+
+   	To register the image, select DockerImage from left menu, under Operator Registry from dashboard and click register button.
+
+
+.. figure:: figures/waterdocker.png
+
+
+3. Now, to create a fogfunction, toggle on Fog Function in the horizontal bar on dashboard. select Fog Function from left menu and click register button. Enter the name and description (optional) as shown. Right click in the workspace and select task and Entity stream from drop down list and configure details.
+
+
+.. figure:: figures/waterfog.png
+
+
+Note: For a details on fogfunction creation follow the `Document link`_ .
+ 
+.. _`Document link`: https://fogflow.readthedocs.io/en/latest/intent_based_program.html
+
+
```

Comment:
Need to reduce the timing of data analytics so that user could observe in 2 minutes.